Foxtrot/Foxtrot/Utils/HelperMethods.cs

```diff
@@ -4186,8 +4186,7 @@ public static Method GetContractMethod(Method instantiatedMethod)
             
             if (candidate != null)
             {
-                if (candidate.Contract != null &&
-                    (candidate.Contract.RequiresCount > 0 || candidate.Contract.EnsuresCount > 0))
+                if (candidate.Contract != null && candidate.Contract.HasAssertions)
```

Comment:
:question: What is `ValidationsCount`? In addition to now checking `AsyncEnsuresCount` here (which I presume fixes the main issue), you now check for `ValidationsCount` as well.